tools/litex-ci-runner/src/main.rs

```diff
@@ -0,0 +1,264 @@
+            .expect("failed to create board flash file")
+            .sync_all()
+            .unwrap();
+    }
+
+    // Now, use tockloader to flash the kernel to the file
+    println!("Flashing kernel \"{}\"...", kernel_binary);
+    let mut tockloader_flash = std::process::Command::new("tockloader")
+        .arg("flash")
+        .arg("--flash-file")
+        .arg(&flash_file.clone().into_os_string())
+        .arg("--board")
+        .arg("litex_sim")
+        .arg("-a")
+        .arg("0x0")
+        .arg(kernel_binary)
+        .spawn()
+        .expect("failed to flash the board kernel");
+    assert!(tockloader_flash.wait().unwrap().success());
+
+    // Use tockloader to install the apps on the board. We expect to have
+    // sufficient builds of the non-relocatable RISC-V apps such that tockloader
+    // can always find a valid combination. If this doesn't work, TOCK_TARGETS
+    // might need some adjustment.
+    for app_path in app_paths.iter() {
+        println!("Installing app \"{}\"...", app_path);
+        let mut tockloader_app_install = std::process::Command::new("tockloader")
+            .current_dir(app_path)
+            .arg("install")
+            .arg("--flash-file")
+            .arg(&flash_file.clone().into_os_string())
+            .arg("--board")
+            .arg("litex_sim")
+            .spawn()
+            .expect("failed to install app in board flash");
+        assert!(tockloader_app_install.wait().unwrap().success());
+    }
+
+    // With the kernel and all apps prepared, run the simulation
+    //
+    // --with-ethernet will cause the simulation to run under a
+    // privileged (root) user. This might change in the future, but
+    // for now will use sudo to get elevated privileges. This will
+    // thus hang if sudo prompts for a password.
+    let mut p = spawn(
+        &format!(
+            "litex_sim \
+            --csr-data-width=32 \
+            --integrated-rom-size=0x100000 \
+            --cpu-variant=secure \
+            --with-ethernet \
+            --timer-uptime \
+            --rom-init {} \
+            --non-interactive",
+            flash_file.to_string_lossy(),
+        ),
+        // The initial build might take a while, which is why we're
+        // allowing it to run for 2min between exp_* commands.
+        Some(120_000),
+    )
+    .expect("failed to run the simulation");
+
+    // Execute user-defined test
+    let res = pty_fn(&mut p);
+
+    // Test completed, kill the simulation
+    kill_litex_sim(&mut p).expect("failed to kill the simulation");
+    p.exp_eof().expect("did not get end of file!");
+
+    Ok(res)
+}
+
+fn run() -> Result<(), Error> {
+    println!("litex-sim CI runner starting...");
+    println!("");
+
+    // Shortcut function, with the kernel binary path and the
+    // libtock-c path prefix fixed and prepended to the
+    // application. Always expects the kernel message first.
+    fn tmpl_run(
```

Comment:
"tmpl"? What is that?

Reply:
That's bad naming. It's a function which performs some basic checks and takes libtock-c app paths relative to the examples directory. We can rename it to "examples_test" or something along those lines.